sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -1168,13 +1168,27 @@ void discard() {
         if (this.bundleCache != null) {
           this.bundleCache.clear();
         }
+        // setupFunction called in createBundleProcessor when BundleProcessorCache.get returns null.
```

Comment:
how about modifying the first sentence to just be:
// setupFunctions are invoked in createBundleProcessor.